### DISCUSSION TITLE
Air-Gap install with private registry for user workloads

### DISCUSSION THREAD

**Discussion by jonbender-stripe** [k3s-io/k3s#4393] [Q&A]
**Describe the solution you'd like**

I'm mainly looking for confirmation that this is possible: Can I install the k3s cluster using the air-gap method but still configure a private registry for pulling any images for k8s workloads submitted to the cluster.

**Additional context**

I'd like to provision k3s without deploying the required images to our private registry and also want to avoid public registries. For all submitted k8s workloads they would use images from an internal registry.

**Backporting**

I'm working on the v1.22 line and do not need backports.

**Comment by brandond** [comment]
Yes, airgap and private registry support are separate features. Airgap tarballs load the image directly into the image store on the node so that it doesn't have to pull them from a registry. Private registry allows you to pull images that don't exist on the node from a local mirror.
